python/mxnet/gluon/nn/basic_layers.py

```diff
@@ -94,8 +123,18 @@ def __repr__(self):
         return s.format(name=self.__class__.__name__,
                         modstr=modstr)
 
-    def __getitem__(self, i):
-        return self._children[i]
+    def __getitem__(self, key):
+        if isinstance(key, integer_types):
+            return self._children[key]
+        elif isinstance(key, py_slice):
+            out = HybridSequential(self._prefix, self._params)
+            for c in self._children[key]:
```

Comment:
registering one layer as children of multiple layers is a bad idea.

Reply:
What's the alternative?

Reply:
Please elaborate

Reply:
return a list or remove this feature

Reply:
why it is problematic

Reply:
With this feature, running a subset of forward will become very easy. Users can simply do
```python
out = model.features[:5](input)
```
I think this simplicity can be useful. If there are problems, let's try to address it together here.